From 45ca88c76c76f011f24abd8dfcd27b533f92c968 Mon Sep 17 00:00:00 2001
From: Gera Shegalov <gera@apache.org>
Date: Mon, 23 Feb 2015 21:05:59 -0800
Subject: [PATCH 2140/2748] HADOOP-11619. FTPFileSystem should override
 getDefaultPort. (Brahma Reddy Battula via gera)

(cherry picked from commit 1dba57271fa56a7383139deb0b89a61c58eedf25)
(cherry picked from commit 62e0e67d497a98844ead5645626953c385609620)

Change-Id: I40106c6b7af98b3d9d61fe6b09638ed32b10b22c
---
 .../org/apache/hadoop/fs/ftp/FTPFileSystem.java    |   10 ++++++
 .../apache/hadoop/fs/ftp/TestFTPFileSystem.java    |   33 ++++++++++++++++++++
 2 files changed, 43 insertions(+)
 create mode 100644 hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/ftp/TestFTPFileSystem.java

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ftp/FTPFileSystem.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ftp/FTPFileSystem.java
index 9d36bcf..fdd2776 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ftp/FTPFileSystem.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ftp/FTPFileSystem.java
@@ -80,6 +80,16 @@ public String getScheme() {
     return "ftp";
   }
 
+  /**
+   * Get the default port for this FTPFileSystem.
+   *
+   * @return the default port
+   */
+  @Override
+  protected int getDefaultPort() {
+    return FTP.DEFAULT_PORT;
+  }
+
   @Override
   public void initialize(URI uri, Configuration conf) throws IOException { // get
     super.initialize(uri, conf);
diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/ftp/TestFTPFileSystem.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/ftp/TestFTPFileSystem.java
new file mode 100644
index 0000000..0ce2a9b
--- /dev/null
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/ftp/TestFTPFileSystem.java
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.fs.ftp;
+
+import org.apache.commons.net.ftp.FTP;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestFTPFileSystem {
+
+  @Test
+  public void testFTPDefaultPort() throws Exception {
+    FTPFileSystem ftp = new FTPFileSystem();
+    Assert.assertEquals(FTP.DEFAULT_PORT, ftp.getDefaultPort());
+  }
+
+}
\ No newline at end of file
-- 
1.7.9.5

